src/Agent.Worker/TaskConditionsEvaluator.cs

```diff
@@ -0,0 +1,50 @@
+using Microsoft.TeamFoundation.DistributedTask.WebApi;
+using Microsoft.VisualStudio.Services.Agent.Util;
+using System;
+using System.Collections.Generic;
+
+namespace Microsoft.VisualStudio.Services.Agent.Worker
+{
+    public static class TaskConditionsEvaluator
+    {
+        public static bool AreConditionsSatisfied(IList<TaskCondition> conditions, Variables variables, IExecutionContext executionContext)
```

Comment:
if private overly complicates the tests, then public static would be fine too. although my gut is the additional setup would not be much.

also, none of the other tests in this repo follow the theory data convention